templates/default/container.cfg.xml.erb

```diff
@@ -1,18 +1,47 @@
 <?xml version="1.0" encoding="UTF-8"?>
-
+<%
+  majorVersion=1
+  minorVersion=1
+  releaseVersion=0
+  deploymentAttributes = ""
+  if !@version.nil?
+    majorVersion=@version.split('.')[0].to_i
+    minorVersion=@version.split('.')[1].to_i
+    releaseVersion=@version.split('.')[2].to_i
+  end
+%>
 <!-- To configure Repose see: http://wiki.openrepose.org/display/REPOSE/Configuration -->
-<% if !@version.nil? && @version.split('.')[0].to_i < 7 %>
+<% if !@version.nil? && majorVersion < 7 %>
 <repose-container xmlns='http://docs.rackspacecloud.com/repose/container/v2.0'>
 <% else %>
 <repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
-<% end %>
-    <deployment-config <% if @content_body_read_limit != nil %>content-body-read-limit="<%= @content_body_read_limit %>"<% end %> connection-timeout="<%= @connection_timeout %>" read-timeout="<%= @read_timeout %>" <% if @client_request_logging != nil %>client-request-logging="<%= @client_request_logging %>"<% end %> <% if @proxy_thread_pool != nil %>proxy-thread-pool="<%= @proxy_thread_pool %>"<% end %>>
-        <deployment-directory auto-clean="<%= @deploy_auto_clean %>">/var/repose</deployment-directory>
-        <artifact-directory check-interval="<%= @filter_check_interval %>">/usr/share/repose/filters</artifact-directory>
-<% if @version.nil? || @version.split('.')[0].to_i > 6 %>
-        <logging-configuration href="file:///etc/repose/log4j2.xml" />
-<% else %>
-        <logging-configuration href="log4j.properties" />
-<% end %>  
-    </deployment-config>
+<%
+  end
+  if @content_body_read_limit != nil
+    deploymentAttributes.concat(" content-body-read-limit=\"").concat(@content_body_read_limit.to_s).concat("\"")
+  end
+  # After version 7.3.8, client_request_logging, connection_timeout, read-timeout & proxy-thread-pool attributes are depricated.
+  # So here we had added check to add these attributes only for version smaller than 7.3.8.
+  if (!@version.nil? &&
+    (majorVersion < 7 ||
```

Comment:
won't the normal string comparison work here? `if version < '7.3.8'`. test ruby code shows it might be an option

```
version='7.3.8'
['1.2.3', '1.2', '7.0', '8.0', '7.4.1', '7.3.9', '7.2.1', '8.0.0'].each do |i|
  if version < i
    puts "#{ version } is less than #{i}" 
  else
    puts "#{ version } is greater than #{i}" 
  end
end

OUTPUT:
7.3.8 is greater than 1.2.3
7.3.8 is greater than 1.2
7.3.8 is greater than 7.0
7.3.8 is less than 8.0
7.3.8 is less than 7.4.1
7.3.8 is less than 7.3.9
7.3.8 is greater than 7.2.1
7.3.8 is less than 8.0.0
```

Reply:
DOne the changes